controllers/admin/AdminSearchController.php

```diff
@@ -470,10 +475,55 @@ public function renderView()
                 $this->tpl_view_vars['modules'] = $this->_list['modules'];
             }
 
+            $this->getSearchPanels($searchedExpression);
+
             return parent::renderView();
         }
     }
 
+    protected function getSearchPanels(string $searchedExpression): void
+    {
+        // Build native search panels
+        $searchPanels = [];
+        $searchPanels[] = new SearchPanel(
+            $this->trans('Search docs.prestashop-project.org', [], 'Admin.Navigation.Search'),
+            $this->trans('Go to the documentation', [], 'Admin.Navigation.Search'),
+            'https://docs.prestashop-project.org/welcome/',
+            [
+                'q' => $searchedExpression,
+            ]
+        );
+
+        // Get additional search panels from hooks
+        $alternativeSearchPanelsFromModules = Hook::exec(
+            'actionGetAlternativeSearchPanels',
+            [
+                'previous_search_panels' => $searchPanels,
+                'bo_query' => $searchedExpression,
+            ],
+            null,
+            true
+        );
+
+        foreach ($alternativeSearchPanelsFromModules as $alternativeSearchPanelsFromModule) {
+            foreach ($alternativeSearchPanelsFromModule as $alternativeSearchPanel) {
+                if ($alternativeSearchPanel instanceof SearchPanelInterface) {
+                    $searchPanels[] = $alternativeSearchPanel;
+                }
+            }
+        }
+
+        // Transform the search panels and inject them to the view
+        $this->tpl_view_vars['searchPanels'] = [];
+        foreach ($searchPanels as $searchPanel) {
+            $this->tpl_view_vars['searchPanels'][] = [
+                'title' => $searchPanel->getTitle(),
+                'button_label' => $searchPanel->getButtonLabel(),
+                'link' => $searchPanel->getLink(),
+            ];
+        }
```

Comment:
Instead of looping twice, maybe you can create a function `getTemplateVars(SearchPanelInterface $searchPanel): array` so in your loop

```php
foreach ($alternativeSearchPanelsFromModules as $alternativeSearchPanelsFromModule) {
    foreach ($alternativeSearchPanelsFromModule as $alternativeSearchPanel) {
        if ($alternativeSearchPanel instanceof SearchPanelInterface) {
            $this->tpl_view_vars['searchPanels'][] = $this->getTemplateVars($alternativeSearchPanel);
        }
    }
}
```

Reply:
You're right @MeKeyCool, I forgot about the first native SearchPanel 😄